### PR TITLE
VZ-4015: explicitly pass appconfig annotations to component workloads

### PR DESCRIPTION
**Pull request by guoyongzhang**
# Description

explicitly pass appconfig "verrazzano.io/restart-version" annotation to component workloads

Fixes VZ-4015

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/appconfig/appconfig_controller.go [hunk 1]
```diff
@@ -97,17 +100,29 @@ func (r *Reconciler) restartComponent(ctx context.Context, componentName, compon
 
 	switch workload.GetKind() {
 	case "VerrazzanoCoherenceWorkload":
-		// "verrazzano.io/restart-version" will be automatically set on VerrazzanoCoherenceWorkload
```
Comment (pfmackin) [inline application-operator/controllers/appconfig/appconfig_controller.go:100]: You need to update the resource after setting the annotations

Comment (mbenmclean) [reply application-operator/controllers/appconfig/appconfig_controller.go:100]: Hmmm it appears to be working but Paul's right you are not updating the resource after. I wonder why this is?

Comment (mbenmclean) [reply application-operator/controllers/appconfig/appconfig_controller.go:100]: Fixed in next commit

File: application-operator/controllers/appconfig/appconfig_controller.go [hunk 2]
```diff
@@ -97,17 +100,29 @@ func (r *Reconciler) restartComponent(ctx context.Context, componentName, compon
...
-		// VerrazzanoCoherenceWorkload reconciler processes the annotation on its own
-		// nothing needs to be done here
+		// pass "verrazzano.io/restart-version" to VerrazzanoCoherenceWorkload
+		err = r.addAnnotation(ctx, workload, restartVersion)
```
Comment (pfmackin) [inline application-operator/controllers/appconfig/appconfig_controller.go:104]: cleaner just to return for all of these cases
return r.addAnnotation...

**Pull request closed without merging**